reddit_service_websockets/socketserver.py

```diff
@@ -128,6 +129,15 @@ def __init__(self, metrics,
         self.quiesced = False
         self.connections = set()
 
+        # register SIGUSR2 to trigger quiescing,
+        #  useful if server processes are behind
+        #  a process manager like einhorn.
+        def _handle_quiesce_signal(_, frame):
+            self._quiesce({}, bypass_auth=True)
+
+        signal.signal(signal.SIGUSR2, _handle_quiesce_signal)
```

Comment:
It feels kinda weird to me to register signal hooks here rather than the app's main function, would it be possible to do it there instead so that it's more obvious reading the app startup flow that this signal's used?